### PR TITLE
add wishlist functionality

### PR DESCRIPTION
**Pull request by Varun-28**
Added wishlist functionality using useContext.

- Now, users can add products to the wishlist
- remove products from wishlist
- cannot add similar products 2 times into the wishlist

wishlist

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -13,7 +14,9 @@ ReactDOM.render(
   <React.StrictMode>
     <BrowserRouter>
       <ProductProvider>
-        <App />
+        <WhishlistProvider>
```
Comment (Hamdrive) [inline src/index.js]: Take care of typos, here it would be `Wishlist`

File: src/utilities/wishlistFun.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
+export const wishlistFun = (wishlistState, action) => {
+    switch(action.type){
+        case ("ADD-TO-WISHLIST"):
+            if(wishlistState.items.filter(item => item._id === action.payload._id).length !== 0){
```
Comment (Hamdrive) [inline src/utilities/wishlistFun.js]: Take care of formatting your code, use a code formatter

**Review by Hamdrive — COMMENTED** [review]
Good progress being made, like the implementation of destructuring when passing props to components

**Review by piyushsh25 — APPROVED** [review]
The code looks good and clean. You can work on suggestions given by the other reviewer.

File: src/index.js (hunk 1 above)
Comment (Varun-28) [reply src/index.js]: updated. ✅

File: src/utilities/wishlistFun.js (hunk 2 above)
Comment (Varun-28) [reply src/utilities/wishlistFun.js]: done ✅

**Pull request merged**